src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -924,6 +981,10 @@ private void checkNeedReset() {
 
     // @GuardedBy("this") // Restore once replacement for JSR-305 available
     private void stateReset() {
+        // While this shouldn't happen, we cover this case to ensure we return to a good state.
+        if (context != null && context.isReleased()) {
```

Comment:
No. But while debugging something else I found a hypothetical (but probably impossible flow) which would have resulted in a non-null released context at the end of a crypto operation. I couldn't even figure out how to trigger it as it would have probably required things breaking within libcrypto. But, I spent so long trying to figure out if it was possible that it seemed easier to simply cover the case so that even if it were to happen we'd still recover properly.